application/common/core.cpp

```diff
@@ -71,11 +71,12 @@ void loop()
     /* Check for events */
     while (SDL_PollEvent(&event))
     {
-        if (event.type == SDL_QUIT)
+        if (event.type == SDL_QUIT ||
+            event.type == SDL_APP_TERMINATING)
```

Comment:
Maybe this would be better suited for the priority filter around line 300: `sdlPriorityEvents`